### PR TITLE
WIP: Test elasticsearch enterprise tests with OpenSearch docker image

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR aims to show Mattermost's compatibility with Open Search, by running the enterprise unit tests with an Open Search container.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-56546

#### Release Note

```release-note
Mattermost server has been unit-tested with Open Search
```

### CODE REVIEW

File: server/Makefile [hunk 1]
```diff
@@ -156,7 +156,9 @@ PLUGIN_PACKAGES += mattermost-plugin-nps-v1.3.3
 PLUGIN_PACKAGES += mattermost-plugin-servicenow-v2.3.4
 PLUGIN_PACKAGES += mattermost-plugin-zoom-v1.6.2
 
-EE_PACKAGES=$(shell $(GO) list $(BUILD_ENTERPRISE_DIR)/...)
+EE_PACKAGES=$(shell $(GO) list $(BUILD_ENTERPRISE_DIR)/elasticsearch)
+export MM_ELASTICSEARCHSETTINGS_USERNAME := admin
+export MM_ELASTICSEARCHSETTINGS_PASSWORD := admin
```
Comment (mickmister) [inline server/Makefile:161]: This is obviously temporary. We will need a way to alternate between ES and OpenSearch in the test environment. AFAIK these credentials are required for OpenSearch. Maybe we can configure both ES and OpenSearch identically to accept the same login creds, or have separate configurations for each test case.

File: server/build/docker-compose.common.yml [hunk 2]
```diff
@@ -87,7 +87,9 @@ services:
       LDAP_DOMAIN: "mm.test.com"
       LDAP_ADMIN_PASSWORD: "mostest"
   elasticsearch:
-    image: "mattermostdevelopment/mattermost-elasticsearch:7.17.10"
+    build:
+      context: .
+      dockerfile: ./Dockerfile.opensearch
```
Comment (mickmister) [inline server/build/docker-compose.common.yml:92]: This is also temporary. We'll need a way to swap out these images/containers to run tests on both containers, or have OpenSearch as a separate container defined in this file, though it cannot run at the same time as the elasticsearch container due to the port mapping. We could change that as well though. Open to what others think is the most maintainable route

**Comment by agnivade** [comment]
To clarify, this is just a sample PR to run the unit tests and not something we actually want to merge right? Not sure if you really wanted code review here.

**Comment by mickmister** [comment]
@agnivade I mainly made this so we can look at the CI output together. But also to get feedback on how we should make this configurable to run tests on both containers in our pipeline

**Comment by agnivade** [comment]
We should discuss first with PM whether we really want to run both containers in our pipeline or not. We have never officially tested with opensearch. And we shouldn't be using the elasticsearch client to work with opensearch backend.

It's fine to do a one-off test and verify the test results, but I think we need to have a deeper discussion if we are considering officially supporting opensearch. This has implications like extra code maintenance, developer training etc.

**Review by fmartingr — COMMENTED** [review]
I've seen the changes and the discussion and I know this is just a test, my only comment here would be to use opensearch as a separate service altogether. That is: it's own container (leave elastic as is), it's own code path, etc.

@agnivade I think this work was only exploratory, to do an ad-hoc run of the specific version opensearch says that is compatible with the elasticsearch APIs to check that it properly works, but after that it would be handed over to product for decissions in how to move forward (correct me if wrong @mickmister)

**Comment by agnivade** [comment]
That's what I thought as well. But in that case, we don't need a PR review or anything.

**Pull request closed without merging**